docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,785 @@
+# MeshTrafficPermission
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4222
+
+## Context and Problem Statement
+
+[New policy matching MADR](./005-policy-matching.md) introduces a new approach how Kuma applies configuration to proxies.
+Rolling out strategy implies creating a new set of policies that satisfy a new policy matching.
+Current MADR aims to define a MeshTrafficPermission.
+
+## Considered Options
+
+* Create a MeshTrafficPermission policy
+
+## Decision Outcome
+
+Chosen option: create a MeshTrafficPermission policy
+
+### Overview
+
+MeshTrafficPermission allows blocking unwanted incoming traffic for the service.
+There might be several reasons why it could be useful to block unwanted traffic:
+* following the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)
+* cut off access for clients that shouldn't use your service (legacy clients, clients from another zone, etc.)
+
+### Specification
+
+Top-level targetRef can have all available kinds:
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset|MeshGatewayRoute|MeshHTTPRoute
+  name: ...
+```
+
+MeshTrafficPermission is an inbound policy that's why it has a "from" array:
+
+```yaml
+from:
+  - targetRef:
+      kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+      name: ...
+```
+
+Configuration for MeshTrafficPermission has a single field "action" with values "ALLOW", "DENY", 
```

Comment:
There is now an action called log - https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/rbac/v3/rbac.proto.html?highlight=action%20log#envoy-v3-api-enum-config-rbac-v3-rbac-action are we considering it in any way?